runtime/binding-filesystem/src/main/java/io/aklivity/zilla/runtime/binding/filesystem/internal/stream/FileSystemStreamFactory.java

```diff
@@ -14,13 +14,15 @@
  */
 package io.aklivity.zilla.runtime.binding.filesystem.internal.stream;
 
+import io.aklivity.zilla.runtime.binding.filesystem.internal.FileSystemWatcher;
 import io.aklivity.zilla.runtime.engine.binding.BindingHandler;
 import io.aklivity.zilla.runtime.engine.config.BindingConfig;
 
 public interface FileSystemStreamFactory extends BindingHandler
 {
     void attach(
-        BindingConfig binding);
+        BindingConfig binding,
+        FileSystemWatcher fileSystemWatcher);
```

Comment:
Feels like it shouldn't be necessary to add a separate parameter to `attach` since you have control over the construction of the implementation of this interface.